src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -691,10 +745,29 @@ def _save_or_select(self, format="binary_folder", folder=None, unit_ids=None) ->
 
         # make a copy of extensions
         # note that the copy of extension handle itself the slicing of units when necessary and also the saveing
-        for extension_name, extension in self.extensions.items():
-            new_ext = new_sorting_analyzer.extensions[extension_name] = extension.copy(
-                new_sorting_analyzer, unit_ids=unit_ids
-            )
+        sorted_extensions = _sort_extensions_by_dependency(self.extensions)
+        recompute_dict = {}
+
+        for extension_name, extension in sorted_extensions.items():
+            if units_to_merge is not None:
+                if merging_mode == "soft":
+                    new_sorting_analyzer.extensions[extension_name] = extension.merge(
+                        new_sorting_analyzer,
+                        units_to_merge=units_to_merge,
+                        new_unit_ids=unit_ids,
+                        keep_mask=keep_mask,
+                        verbose=verbose,
+                        **job_kwargs,
+                    )
+                elif merging_mode == "hard":
+                    recompute_dict[extension_name] = extension.params
+            else:
+                new_sorting_analyzer.extensions[extension_name] = extension.copy(
+                    new_sorting_analyzer, unit_ids=unit_ids
+                )
```

Comment:
We need to capture the params fore the recompute. no ?
recompute_dict is not update  no ?

Reply:
recompute_dict is filled with the params of the extension at the time of the merge